test/spec.yml

```diff
@@ -1373,3 +1374,346 @@
+        EOF
+    - in: bn deploy testupdatetriggerfunc2
+      out: |-
+        Deployed function testupdatetriggerfunc2
+        Invoke with one of:
+          "bn invoke testupdatetriggerfunc2"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert testupdatetrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func testupdatetriggerfunc
+      out: CREATED
+    - in: bn trigger upsert testupdatetrigger --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func testupdatetriggerfunc2
+      out: UPDATED
+
+- test: Test 2 new triggers, same source, different target (good-path)
+  cleanup:
+    - bn trigger remove test2triggers
+    - bn trigger remove test2triggers2
+    - bn remove test2triggersfunc
+    - bn remove test2triggersfunc2
+  steps:
+    - in: bn create node8 test2triggersfunc
+      out: |-
+          Created function * in /home/dockeruser/test
+            (use "bn deploy test2triggersfunc" to deploy the function)
+    - in: bn deploy test2triggersfunc
+      out: |-
+        Deployed function test2triggersfunc
+        Invoke with one of:
+          "bn invoke test2triggersfunc"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: |-
+        cat <<EOF >/home/dockeruser/test/binaris.yml
+        functions:
+          test2triggersfunc:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+          test2triggersfunc2:
+            file: function.js
+            entrypoint: handler
+            executionModel: concurrent
+            runtime: node8
+        EOF
+    - in: bn deploy test2triggersfunc2
+      out: |-
+        Deployed function test2triggersfunc2
+        Invoke with one of:
+          "bn invoke test2triggersfunc2"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert test2triggers --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func test2triggersfunc
+      out: CREATED
+    - in: bn trigger upsert test2triggers2 --source-uri "redis+tls://:${REDIS_PASS}@misc-redis.binaris.io/0#numbers" --target-func test2triggersfunc2
+      out: CREATED
+
+- test: Test new trigger with bad source uri (bad-path)
+  cleanup:
+    - bn remove testbadtriggerfunc
+  steps:
+    - in: bn create node8 testbadtriggerfunc
+      out: |-
+          Created function * in /home/dockeruser/test
+            (use "bn deploy testbadtriggerfunc" to deploy the function)
+    - in: bn deploy testbadtriggerfunc
+      out: |-
+        Deployed function testbadtriggerfunc
+        Invoke with one of:
+          "bn invoke testbadtriggerfunc"
+          "curl -H X-Binaris-Api-Key:* https://*"
+    - in: bn trigger upsert testbadtrigger --source-uri redis://not-a-valid-redis-uri --target-func testbadtriggerfunc
+      exit: 1
+      err: |-
+        RequestId: *
+        Internal Binaris server error*
```

Comment:
Not the expected error here.

Reply:
yet it passes...
this needs to be updated on `spice`

Reply:
You should get the correct error now